### PR TITLE
feat(translation): translate readability part chapter one

### PR DESCRIPTION
**Pull request by natromankevich**
**Yes, I promise I've read the [Contributions Guidelines](https://github.com/getify/Functional-Light-JS/blob/master/CONTRIBUTING.md)**

Add translation for chapter I, part "Readability".

### CODE REVIEW

File: manuscript/ch1.md [hunk 1]
```diff
@@ -128,48 +128,48 @@ function constructMsg(v) { return `The magic number is: ${v}`; }
 
 ## Читабельность
 
-Readability is not a binary characteristic. It's a largely subjective human factor describing our relationship to code. And it will naturally vary over time as our skills and understanding evolve. I have experienced effects similar to the following figure, and anecdotally many others I've talked to have as well.
+Читабельность - это не бинарная характеристика. Это в значительной степени субъективный человеческий фактор, описывающий наше отношение к коду. И это, естественно, будет меняться со временем по мере развития наших навыков и понимания. У меня был опыт подобный показанному на нижеприведённом графике, и, как ни странно, многие другие, с которыми я разговаривал, испытывали тоже самое.
```
Comment (fxzhukov) [inline manuscript/ch1.md]:
```suggestion
Читабельность - это не бинарная характеристика. Это в значительной степени субъективный человеческий фактор, описывающий наше отношение к коду. И он, естественно, будет меняться со временем по мере развития наших навыков и понимания. У меня был опыт подобный показанному на нижеприведённом графике, и, как ни странно, многие другие, с которыми я разговаривал, испытывали тоже самое.
```

File: manuscript/ch1.md [hunk 2]
```diff
@@ -128,48 +128,48 @@ function constructMsg(v) { return `The magic number is: ${v}`; }
...
 
 <p align="center">
     <img src="images/fig17.png" width="50%">
 </p>
 
-You may just find yourself experiencing similar effects as you work through the book. But take heart; if you stick this out, the curve comes back up!
+Возможно, вы будете испытывать похожее во время прочтения этой книги. Однако наберитесь мужества. Если вы продержитесь до конца, кривая снова пойдёт вверх!
```
Comment (fxzhukov) [inline manuscript/ch1.md]:
```suggestion
Возможно, вы будете испытывать похожие ощущения во время прочтения этой книги. Однако наберитесь мужества. Если вы продержитесь до конца, кривая снова пойдёт вверх!
```

File: manuscript/ch1.md [hunk 3]
```diff
@@ -128,48 +128,48 @@ function constructMsg(v) { return `The magic number is: ${v}`; }
...
 
-*Imperative* describes the code most of us probably already write naturally; it's focused on precisely instructing the computer *how* to do something. Declarative code -- the kind we'll be learning to write, which adheres to FP principles -- is code that's more focused on describing the *what* outcome.
+*Императивный стиль программирования*  описывает код, который большинство из нас, вероятно, уже пишут естественным образом. Он нацелен на то, чтобы точно давать инструкции компьютору *как* что-то выполнить. Декларативный код - тот, который мы будем учиться писать, следует принципам ФП - это код, который больше ориентирован на описание результата *что*.
```
Comment (fxzhukov) [inline manuscript/ch1.md]:
```suggestion
*Императивный стиль программирования*  описывает код, который большинство из нас пишет каждый день. Он нацелен на то, чтобы точно давать инструкции компьютору *как* что-то выполнить. Декларативный код - тот, который мы будем учиться писать, следует принципам ФП - это код, который больше ориентирован на описание того, **чего** мы хотим добиться.
```

File: manuscript/ch1.md [hunk 4]
```diff
@@ -128,48 +128,48 @@ function constructMsg(v) { return `The magic number is: ${v}`; }
...
 
-I know asking you to believe that at this point is a leap of faith.
+Я знаю, просить вас поверить в это на данный момент - это вопрос доверия.
 
-It takes a lot more effort, and sometimes more code, to improve its readability as I'm suggesting, and to minimize or eliminate many of the mistakes that lead to bugs. Quite honestly, when I started writing this book, I could never have written (or even fully understood!) that second snippet. As I'm now further along on my journey of learning, it's more natural and comfortable.
+Требуется гораздо больше усилий, а иногда и больше кода, чтобы улучшить его читабельность как я предлагаю, и свести к минимуму или устранить многие ошибки, которые приводят к багам. Честно говоря, когда я начал написание этой книги, я никогда бы не смог бы написать (или даже полностью понять!) этот второй фрагмент. Сейчас, дальше на моём пути обучения, это становится более естественным и удобным.
```
Comment (fxzhukov) [inline manuscript/ch1.md]:
```suggestion
Требуется гораздо больше усилий, а иногда и больше кода, чтобы улучшить его читабельность как я предлагаю, и свести к минимуму или устранить многие ошибки, которые приводят к багам. Честно говоря, когда я начал написание этой книги, я никогда бы не смог бы написать (или даже полностью понять!) этот второй фрагмент. Сейчас, когда я более знаком с функциональным подходом, он выглядит даже более естественным и удобным. 
```

File: manuscript/ch1.md [hunk 5]
```diff
@@ -128,48 +128,48 @@ function constructMsg(v) { return `The magic number is: ${v}`; }
...
+Требуется гораздо больше усилий, а иногда и больше кода, чтобы улучшить его читабельность как я предлагаю, и свести к минимуму или устранить многие ошибки, которые приводят к багам. Честно говоря, когда я начал написание этой книги, я никогда бы не смог бы написать (или даже полностью понять!) этот второй фрагмент. Сейчас, дальше на моём пути обучения, это становится более естественным и удобным. 
 
-If you're hoping that FP refactoring, like a magic silver bullet, will quickly transform your code to be more graceful, elegant, clever, resilient, and concise -- that it will come easy in the short term -- unfortunately that's just not a realistic expectation.
+Если вы надеетесь, что ФП рефакторинг, подобно волшебной палочке, быстро преобразует ваш код, сделав его более изящным, элегантным, умным, устойчивым и лаконичным - что в краткосрочной перспективе это будет легко - к сожалению, это просто нереалистичное ожидание.
```
Comment (fxzhukov) [inline manuscript/ch1.md]:
```suggestion
Если вы надеетесь, что рефакторинг вашего кода в функциональном стиле, подобно волшебной палочке, быстро преобразит ваш код, сделав его более изящным, элегантным, умным, устойчивым и лаконичным - что в краткосрочной перспективе это будет легко - к сожалению, это просто нереалистичное ожидание. 
```

File: manuscript/ch1.md [hunk 6]
```diff
@@ -128,48 +128,48 @@ function constructMsg(v) { return `The magic number is: ${v}`; }
...
 
-If you're hoping that FP refactoring, like a magic silver bullet, will quickly transform your code to be more graceful, elegant, clever, resilient, and concise -- that it will come easy in the short term -- unfortunately that's just not a realistic expectation.
+Если вы надеетесь, что ФП рефакторинг, подобно волшебной палочке, быстро преобразует ваш код, сделав его более изящным, элегантным, умным, устойчивым и лаконичным - что в краткосрочной перспективе это будет легко - к сожалению, это просто нереалистичное ожидание. 
 
-FP is a very different way of thinking about how code should be structured, to make the flow of data much more obvious and to help your reader follow your thinking. It will take time. This effort is eminently worthwhile, but it can be an arduous journey.
+ФП - это совсем другой тип мышления о том, как должен быть структурирован код, чтобы сделать поток данных намного более очевидным и помочь вашему читателю следовать вашим мыслям. Это займёт время. Это усилие в исключительной степени стоит того, но это может быть трудоёмким процессом.
```
Comment (fxzhukov) [inline manuscript/ch1.md]:
```suggestion
ФП - это совсем другой тип мышления о том, как должен быть структурирован код, чтобы сделать работу вашей программы с данными намного более очевидной и помочь читателям вашего кода следовать вашим мыслям. Это займёт время, но приложенные усилия в исключительной степени стоят достигнутого результата.
```

File: manuscript/ch1.md [hunk 7]
```diff
@@ -128,48 +128,48 @@ function constructMsg(v) { return `The magic number is: ${v}`; }
...
 
-It still often takes me multiple attempts at refactoring a snippet of imperative code into more declarative FP, before I end up with something that's clear enough for me to understand later. I've found converting to FP is a slow iterative process rather than a quick binary flip from one paradigm to another.
+Даже сейчас мне часто требуется несколько попыток рефакторинга фрагмента императивного кода в более декларативный ФП, прежде чем я получу что-то достаточно ясное, чтобы я мог понять это позже. Я обнаружил, что преобразование в ФП - это медленный итеративный процесс, а не быстрый бинарный переход от одной парадигмы к другой.
```
Comment (fxzhukov) [inline manuscript/ch1.md:166]:
```suggestion
Даже сейчас мне часто требуется несколько попыток рефакторинга фрагмента императивного кода в более декларативный ФП, прежде написанный код будет достаточно ясным, чтобы я мог понять его позже. Я обнаружил, что погружение в ФП - это медленный итеративный процесс, а не быстрый бинарный переход от одной парадигмы к другой.
```

File: manuscript/ch1.md [hunk 8]
```diff
@@ -128,48 +128,48 @@ function constructMsg(v) { return `The magic number is: ${v}`; }
...
 
-I also apply the "teach it later" test to every piece of code I write. After I've written a piece of code, I leave it alone for a few hours or days, then come back and try to read it with fresh eyes, and pretend as if I need to teach or explain it to someone else. Usually, it's jumbled and confusing the first few passes, so I tweak it and repeat!
+Я также применяю тест "научу этому позже" к каждому фрагменту кода, который я пишу. После написания части кода, я оставляю его в покое на несколько часов или дней, затем возвращаюсь и пытаюсь читать его со свежим взглядом, и представляю, что мне нужно научить или объяснить это кому-то другому. Обычно первые несколько попыток получаются сумбурными и сбивающими с толку, поэтому я подправляю их и повторяю процесс!
```
Comment (fxzhukov) [inline manuscript/ch1.md:168]:
```suggestion
Я также применяю тест "прочти это позже" к каждому фрагменту кода, который я пишу. После написания части кода, я оставляю его в покое на несколько часов или дней, затем возвращаюсь и пытаюсь читать его со свежим взглядом, и представляю, что мне нужно изучить его или объяснить кому-то другому. Обычно первые несколько попыток получаются сумбурными и сбивающими с толку, поэтому я подправляю код и повторяю процесс!
```

File: manuscript/ch1.md [hunk 9]
```diff
@@ -128,48 +128,48 @@ function constructMsg(v) { return `The magic number is: ${v}`; }
...
 
-I'm not trying to dampen your spirits. I really want you to hack through these weeds. I am glad I did it. I can finally start to see the curve bending upward toward improved readability. The effort has been worth it. It will be for you, too.
+Я не пытаюсь ослабить ваш дух. Я действительно хочу, чтобы вы преодолели эти трудности. Я очень рад, что я сделал это. Наконец-то я могу увидить кривую, стремящуюся к улучшению читабельности. Усилия того стоили. Это будет также и для вас.
```
Comment (fxzhukov) [inline manuscript/ch1.md]:
```suggestion
Я не пытаюсь ослабить ваш дух. Я действительно хочу, чтобы вы преодолели эти трудности. Я очень рад, что я сделал это. Я наконец-то могу увидеть кривую, стремящуюся к улучшению читабельности. Мои усилия оправдали себя. Также будет и для вас.
```

**Review by fxzhukov — COMMENTED** [review]
Thank you for your contribution! 🥇 
Left some comments.

File: manuscript/ch1.md (hunk 8 above)
Comment (natromankevich) [reply manuscript/ch1.md:168]: Thanks so much for your feedback @fxzhukov ! I think that "научи этому позже" is more accurate in this case for "teach it later". "Teach"  means "научить кого-то, преподавать кому-то". Also after that, the author says, among other things, about the necessity to explain what you learn to someone else to remember it better., which I see as key advice here.

File: manuscript/ch1.md (hunk 7 above)
Comment (natromankevich) [reply manuscript/ch1.md:166]: Thank you! Missing "чем" after "прежде"? That will be "прежде чем". What do you think?

**Pull request merged**